### PR TITLE
Added Merge engine

### PR DESCRIPTION
**Pull request by M1hacka**

### CODE REVIEW

File: src/infi/clickhouse_orm/engines.py [hunk 1]
```diff
@@ -128,3 +130,27 @@ def create_table_sql(self, db_name):
...
+class Merge(Engine):
+    """
+    The Merge engine (not to be confused with MergeTree) does not store data itself,
+    but allows reading from any number of other tables simultaneously.
+    Writing to a table is not supported
+    https://clickhouse.yandex/docs/en/single/index.html#document-table_engines/merge
+    """
+
+    def __init__(self, table_regex):
+        assert isinstance(table_regex, six.string_types), "'db_name' parameter must be string"
```
Comment (ishirav) [inline src/infi/clickhouse_orm/engines.py]: error message is wrong

File: src/infi/clickhouse_orm/database.py [hunk 2]
```diff
@@ -75,16 +75,16 @@ def create_table(self, model_class):
         Creates a table for the given model class, if it does not exist already.
         '''
         # TODO check that model has an engine
-        if model_class.readonly:
-            raise DatabaseException("You can't create read only table")
+        if model_class.system:
```
Comment (ishirav) [inline src/infi/clickhouse_orm/database.py]: Can you please explain why you stopped checking if the model is readonly?

Comment (M1hacka) [reply src/infi/clickhouse_orm/database.py]: I've divided readonly flag into 2 flags: readonly and system. Before this, readonly flag restricted dropping and creating tables (it was used in system models, and it was suitable for them). But now I've added Merge engine. Table with this engine can be created or dropped, but INSERTs are restricted. So I needed a mechanism to control these restrictions separatly: now readonly flag controls INSERTs only, and system flag controls both INSERTs and CREATE/DROP TABLE.

File: src/infi/clickhouse_orm/engines.py (hunk 1 above)
Comment (M1hacka) [reply src/infi/clickhouse_orm/engines.py]: Fixed.

File: src/infi/clickhouse_orm/fields.py [hunk 3]
```diff
@@ -16,19 +16,21 @@ class Field(object):
     class_default = 0
     db_type = None
 
-    def __init__(self, default=None, alias=None, materialized=None):
+    def __init__(self, default=None, alias=None, materialized=None, readonly=None):
```
Comment (ishirav) [inline src/infi/clickhouse_orm/fields.py]: Some of the Field subclasses override __init__, so you should add the readonly argument there too

Comment (M1hacka) [reply src/infi/clickhouse_orm/fields.py]: Fixed.

**Comment by ishirav** [comment]
Merged and released in v0.9.7

**Pull request merged**